src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -526,4 +530,46 @@ object AppDefinition {
       }
     }
   }
+
+  def residentUpdateIsValid(from: AppDefinition): Validator[AppDefinition] = {
+    def changeNoVolumes: Validator[AppDefinition] = new Validator[AppDefinition] {
+      override def apply(to: AppDefinition): Result = {
+        val fromVolumes = from.persistentVolumes
+        val toVolumes = to.persistentVolumes
+        def sameSize = fromVolumes.size == toVolumes.size
+        def noChange = from.persistentVolumes.forall { fromVolume =>
+          toVolumes.find(_.containerPath == fromVolume.containerPath).contains(fromVolume)
+        }
+        if (sameSize && noChange) Success
+        else Failure(Set(RuleViolation(to, "Persistent volumes can not be changed!", None)))
```

Comment:
Just speaking my mind: Originally, we decided to call these `Local Volumes`. However, the underlying Mesos functionality is `Persistent Volumes` and I begin to feel that renaming these will only lead to problems. My proposal: leave this here as it is. We should make sure in a cleanup phase that we have consistent naming (see `LocalVolumeId` as an example ...)